pkg/extension/sumologicextension/extension.go

```diff
@@ -57,40 +61,60 @@ const (
 )
 
 func newSumologicExtension(conf *Config, logger *zap.Logger) (*SumologicExtension, error) {
+	collectorName := ""
+	uuid := uuid.New()
+	hostname, err := os.Hostname()
+	if err != nil {
+		return nil, err
+	}
+	credentialsGetter := credsGetter{
+		conf:   conf,
+		logger: logger,
+	}
 	if conf.CollectorName == "" {
-		return nil, errors.New("collector name is unset")
+		key := fmt.Sprintf("%s%s%s", conf.CollectorName, conf.Credentials.AccessID, conf.Credentials.AccessKey)
+		// If collector name is not set by the user check if the collector was restarted
+		if !credentialsGetter.CheckCollectorCredentials(key) {
+			// If credentials file is not stored on filesystem generate collector name
+			collectorName = fmt.Sprintf("%s-%s", hostname, uuid.String())
+		}
```

Comment:
I believe we need to account for more facts here: if user specifies `clobber` as `true` then do we really want to append uuid? Doesn't make sense to me, the user specified by the clobber flag: 

> Please override that collector name for me, whatever it is. (if it's empty, override the default name that corresponds to empty collector name in the config)